public/app/features/query/state/queryAnalytics.ts

```diff
@@ -31,21 +66,11 @@ export function emitDataRequestEvent(datasource: DataSourceApi) {
     const totalQueries = Object.keys(queryCacheStatus).length;
     const cachedQueries = Object.values(queryCacheStatus).filter((val) => val === true).length;
 
-    const eventData: DataRequestEventPayload = {
-      eventName: MetaAnalyticsEventName.DataRequest,
-      datasourceName: datasource.name,
-      datasourceId: datasource.id,
-      datasourceUid: datasource.uid,
-      datasourceType: datasource.type,
-      panelId: data.request.panelId,
-      dashboardId: data.request.dashboardId,
-      dataSize: 0,
-      duration: data.request.endTime! - data.request.startTime,
-      totalQueries,
-      cachedQueries,
-    };
+    eventData.panelId = data.request!.panelId;
```

Comment:
nit: I'm aware this isn't used anywhere else right now but maybe we should future proof this to prevent runtime errors caused by `request` not existing? Maybe return an empty string? 🤷

Reply:
@jackw this is where my frontend skills are a bit lacking. There is a guard at the start of the main function `if (!data.request || done) {` - is it too risky to rely on that?

Reply:
Nah not too risky. More a concern for the future if something else wants to use the same function. Only a nit - let's not worry about it for now. 👍